lidlplus/api.py

```diff
@@ -252,3 +253,21 @@ def ticket(self, ticket_id):
         kwargs = {"headers": self._default_headers(), "timeout": self._TIMEOUT}
         url = f"{self._TICKET_API}/{self._country}/tickets"
         return requests.get(f"{url}/{ticket_id}", **kwargs).json()
+
+    def coupons(self):
+        """Get list of all coupons"""
+        url = f"{self._COUPONS_API}/v2/{self._country}"
+        kwargs = {"headers": self._default_headers(), "timeout": self._TIMEOUT}
+        return requests.get(url, **kwargs).json()
+
+    def activate_coupon(self, coupon_id):
+        """Activate single coupon by id"""
+        url = f"{self._COUPONS_API}/v1/{self._country}/{coupon_id}/activation"
+        kwargs = {"headers": self._default_headers(), "timeout": self._TIMEOUT}
+        return requests.post(url, **kwargs).json()
+
+    def deactivate_coupon(self, coupon_id):
+        """Deactivate single coupon by id"""
+        url = f"{self._COUPONS_API}/v1/{self._country}/{coupon_id}/activation"
```

Comment:
Is this the same endpoint like the activation? I think it would be better than to have a `_switch_coupon_activation` function that is called by both functions

Reply:
Sorry! I oversee that this was a different method (post and delete) :see_no_evil: 
Than this is just too much overhead, I would revert the commit and use your initial solution, sorry for the extra work :grimacing:

Reply:
Oh yeah, no worries it's a simple change, I would have just fixed that 😅